File: flygandefika/submissions/accepted/joshua.cpp
#include <bits/stdc++.h>

using namespace std;


int main()
{
    int n, m;
    cin >> n >> m;

    vector<int> controls(n);
    vector<int> sources(m);

    for (int i = 0; i < n; i++) cin >> controls[i];
    for (int i = 0; i < m; i++) cin >> sources[i];

    vector<pair<int, int>> ans(n, { -1, int(2e9) });

    for (int i = 0; i < n; i++)
    {
        auto it = lower_bound(begin(sources), end(sources), controls[i]);
        if (it != end(sources)) ans[i] = { it - begin(sources), abs(controls[i] - *it) };
        if (it != begin(sources) && abs(controls[i] - *prev(it)) < ans[i].second) ans[i] = { prev(it) - begin(sources), abs(controls[i] - *prev(it)) };

    }

    for (int i = 0; i < n; i++)
    {
        cout << ans[i].first << " ";
    }

    cout << "\n";

    return 0;
}

File: flygandefika/submissions/accepted/joshua.py
#!/usr/bin/python3

n,m = [int(i) for i in input().split()]

controls = sorted([int(i) for i in input().split()])
sources = sorted([int(i) for i in input().split()])

ans = [[-1,int(2e9)] for i in range(n)]

i = 0
j = 0


# To the right
while True:
    #print(i,j)
    if abs(controls[i]-sources[j])<ans[i][1]:
            ans[i] = [j,abs(controls[i]-sources[j])]
        
    if controls[i]>sources[j] and i+1<n:
        i+=1
    elif j+1<m:
        j+=1
    elif i+1<n:
        i+=1
    else:
        break

    
i = 0
j = 0

# To the left
while True:
    #print(i,j)
    if abs(controls[i]-sources[j])<ans[i][1]:
            ans[i] = [j,abs(controls[i]-sources[j])]
            
    if controls[i]<sources[j] and i+1 < n:
        i+=1
        
    elif j+1<m:
        j+=1
    elif i+1<n:
        i+=1
    else:
        break


for i in range(n):
    print(ans[i][0], end=" ")
print("")

